src/scale.c

```diff
@@ -180,3 +182,33 @@ avifBool avifImageScale(avifImage * image,
 }
 
 #endif
+
+avifResult avifImageCopySamples(avifImage * dstImage, const avifImage * srcImage, avifPlanesFlags planes)
+{
+    for (int c = AVIF_CHAN_Y; c <= AVIF_CHAN_A; ++c) {
+        if ((!(planes & AVIF_PLANES_YUV) && (c != AVIF_CHAN_A)) || (!(planes & AVIF_PLANES_A) && (c == AVIF_CHAN_A))) {
+            continue;
+        }
+        const uint32_t planeWidth = avifImagePlaneWidth(srcImage, c);
+        const uint32_t planeHeight = avifImagePlaneHeight(srcImage, c);
+        const uint8_t * srcRow = avifImagePlane(srcImage, c);
+        uint8_t * dstRow = avifImagePlane(dstImage, c);
+        const uint32_t srcRowBytes = avifImagePlaneRowBytes(srcImage, c);
+        const uint32_t dstRowBytes = avifImagePlaneRowBytes(dstImage, c);
+        if ((planeWidth != avifImagePlaneWidth(dstImage, c)) || (planeHeight != avifImagePlaneHeight(dstImage, c)) ||
+            (!srcRow != !dstRow) || (srcImage->depth != dstImage->depth)) {
```

Comment:
Nit: Test `srcImage->depth != dstImage->depth` outside the for loop. This check is loop invariant.